editoast/src/client/redis_config.rs

```diff
@@ -1,14 +1,69 @@
+use crate::error::Result;
 use clap::Args;
 use derivative::Derivative;
+use editoast_derive::EditoastError;
+use thiserror::Error;
+use url::Url;
 
 #[derive(Args, Debug, Derivative, Clone)]
 #[derivative(Default)]
 pub struct RedisConfig {
     #[derivative(Default(value = "false"))]
     #[clap(long, env, default_value_t = false)]
     pub is_cluster_client: bool,
-    #[derivative(Default(value = r#""redis://localhost:6379".into()"#))]
-    #[arg(long, env, default_value = "redis://localhost:6379")]
-    /// Redis url like `redis://[:PASSWORD@]HOST[:PORT][/DATABASE]`
-    pub redis_url: String,
+    #[derivative(Default(value = r#""localhost".into()"#))]
+    #[arg(long, env, default_value = "localhost")]
+    pub redis_host: String,
+    #[derivative(Default(value = "6379"))]
+    #[arg(long, env, default_value_t = 6379)]
+    pub redis_port: u16,
+    pub redis_database: Option<String>,
+    pub redis_username: Option<String>,
+    pub redis_password: Option<String>,
```

Comment:
You should keep the same configuration for redis (i.e. a single `redis_url` parameter). This will save us from having to modify the deployment. You can always parse the URL and check its validity.

Reply:
Done.